pkg/controllers/resources/storageclasses/host_syncer.go

```diff
@@ -64,6 +71,12 @@ func (s *hostStorageClassSyncer) SyncToVirtual(ctx *synccontext.SyncContext, eve
 }
 
 func (s *hostStorageClassSyncer) Sync(ctx *synccontext.SyncContext, event *synccontext.SyncEvent[*storagev1.StorageClass]) (_ ctrl.Result, retErr error) {
+	if !ctx.Config.Sync.FromHost.StorageClasses.Selector.Matches(event.Host) {
+		s.EventRecorder().Eventf(event.Virtual, "Warning", "SyncWarning", "did not sync storage class %q because it does not match the selector under 'sync.fromHost.storageClasses.selector'", event.Virtual.Name)
+		s.EventRecorder().Eventf(event.Host, "Warning", "SyncWarning", "did not sync storage class %q because it does not match the selector under 'sync.fromHost.storageClasses.selector'", event.Host.Name)
```

Comment:
We cannot use event.Host here as the event recorder only works for the virtual object